tests/test_units.py

```diff
@@ -20,203 +20,221 @@
+        unitstore.add_unit('newunit', 'second * 2')
+        assert (unitstore.is_unit_defined('newunit') is True)
+
+    def test_add_unit_1(self):
+        """Tests Units.add_unit()."""
+        unitstore = UnitStore()
+        assert (unitstore.is_unit_defined('newunit') is False)
+
+        # Add a new unit called 'newunit' which is millimetres
+        unitstore.add_unit('newunit', 'metre / 1000')
+        assert (unitstore.is_unit_defined('newunit') is True)
+
+    def test_add_unit_3(self):
+        """ Tests Units.add_unit() function. """
+        unitstore = UnitStore()
+        assert (unitstore.is_unit_defined('newunit') is False)
+
+        # Add a new unit called 'newunit' which is metre per second
+        unitstore.add_unit('newunit', 'meter / second')
+        assert (unitstore.is_unit_defined('newunit') is True)
+
+    def test_add_unit_existing(self):
+        """ Tests exception for Units.add_unit() function when unit already exists. """
+        unitstore = UnitStore()
         with pytest.raises(AssertionError):
-            unitstore.add_custom_unit('second', unit_attributes)
-            pytest.fail("Cannot redefine CellML unit <second>")
-            pass
-
-    def test_is_unit_defined(self, quantity_store):
-        """ Tests Units._is_unit_defined() function. """
-        assert (quantity_store._is_unit_defined('ms') is True)
-        assert (quantity_store._is_unit_defined('not_a_unit') is False)
-
-    def test_make_pint_unit_definition(self, quantity_store):
-        """ Tests Units._make_pint_unit_definition() function. """
-        unit_attributes = [{'prefix': -3, 'units': 'metre'},
-                           {'prefix': 'milli', 'exponent': -1, 'units': 'second'},
-                           {'multiplier': 2, 'units': 'kilograms'}
-                           ]
-        assert(quantity_store._make_pint_unit_definition('kg_mm_per_ms', unit_attributes) ==
-               'kg_mm_per_ms=(meter * 1e-3)*(((second * 0.001))**-1)*(2 * kilogram)')
+            unitstore.add_unit('second', 'second * 2')
+            pytest.fail('Cannot redefine CellML unit <second>')
+
+    # def test_is_unit_defined(self, unit_store):
+    #     """ Tests Units._is_unit_defined() function. """
+    #     assert (unit_store.is_unit_defined('ms') is True)
+    #    assert (unit_store.is_unit_defined('not_a_unit') is False)
+
+    # def test_make_pint_unit_definition(self, unit_store):
+    #    """ Tests Units._make_pint_unit_definition() function. """
+    #    unit_attributes = [{'prefix': -3, 'units': 'metre'},
+    #                       {'prefix': 'milli', 'exponent': -1, 'units': 'second'},
+    #                       {'multiplier': 2, 'units': 'kilograms'}
+    #                       ]
+    #    assert(unit_store._make_pint_unit_definition('kg_mm_per_ms', unit_attributes) ==
+    #           'kg_mm_per_ms=(meter * 1e-3)*(((second * 0.001))**-1)*(2 * kilogram)')
+
+    '''
```

Comment:
I'm guessing this bit is just work in progress.

Reply:
I'm leaving these out unless they become public in the future. Testing should happen via the public API,